addons/explosives/functions/fnc_dialPhone.sqf

```diff
@@ -28,22 +26,35 @@ private _arr = [];
 for "_i" from 1 to _ran do {
     _arr = _arr + ['.','..','...',''];
 };
+
+private _explosive = [_code] call FUNC(getSpeedDialExplosive);
+
+private _explosiveType = typeOf (_explosive select 0);
+private _ringtone = GVAR(iedRingtonesHash) getOrDefault [_explosiveType,getArray (configFile >> "CfgAmmo" >> _explosiveType >> QGVAR(ringtones)),true];
+if (_ringtone isEqualTo []) then { _ringtone = [[QPATHTOF(Data\Audio\Cellphone_Ring.wss), 0.75]] };
+//support random ringtones
+_ringtone = selectRandom _ringtone;
+_ringtone params ["_ringtonePath", "_ringtoneLenght"];
+
+//Calculate the ringtone duration before detonation with slight randomness
+private _ringtoneDuration = round(count _arr - (_ringtoneLenght + (random 1.5)- 0.75) / 0.25) max 4;
+
+TRACE_2("ringtone",_ringtone,_ringtoneDuration);
 if (_unit == ace_player) then {
     ctrlSetText [1400,"Calling"];
-    [LINKFUNC(dialingPhone), 0.25, [_unit,4,_arr,_code]] call CALLSTACK(CBA_fnc_addPerFrameHandler);
+    [LINKFUNC(dialingPhone), 0.25, [_unit,4,_arr,_explosive,_ringtonePath,_ringtoneDuration]] call CALLSTACK(CBA_fnc_addPerFrameHandler);
 } else {
-    private _explosive = [_code] call FUNC(getSpeedDialExplosive);
     if ((count _explosive) > 0) then {
+        private _waitTime = 0.25 * _ringtoneDuration;
         [{
-            params ["_unit", "_item"];
-
+            params ["_unit", "_item", "_ringtonePath"];
             if ([_unit, -1, (_item # 0), (_item # 2), "ACE_Cellphone"] call FUNC(checkDetonateHandlers)) then {
-                playSound3D [QUOTE(PATHTO_R(Data\Audio\Cellphone_Ring.wss)), objNull, false, (getPosASL (_item # 0)), 3.16228, 1, 75];
+                playSound3D [_ringtonePath, objNull, false, (getPosASL (_item # 0)), 3.16228, 1, 75];
```

Comment:
If we are adding the ability to play custom sound files, then adding the pitch, volume and distance to the config should be an option.

Reply:
Added this